python/pyspark/sql/functions.py

```diff
@@ -1450,6 +1471,8 @@ def hash(*cols):
 # ---------------------- String/Binary functions ------------------------------
 
 _string_functions = {
+    'upper': 'Converts a string expression to upper case.',
+    'lower': 'Converts a string expression to upper case.',
```

Comment:
This will give the incorrect `since` in the documentation. These functions were introduced in version 1.3.

Reply:
It was already 1.5 before your PR. Shall we don't target it here?

Reply:
You mean keeping as it was after my PR or before it? Because the PR at least made it consistent with the Scala API documentation, whereas this is a form of regression. If the correct version is indeed 1.0, then I'd suggest either including these functions in the `_functions_with_column` list, or creating a separate `_string_functions` for the ones that have actually been there since 1.5.

Reply:
I mean I just intent to leave it as was before your PR.

Before your PR:

```
Help on function lower in module pyspark.sql.functions:

lower(col)
    Converts a string column to lower case.

    .. versionadded:: 1.5
```

After your PR:

```
Help on function lower in module pyspark.sql.functions:

lower(col)
    Converts a string expression to upper case.

    .. versionadded:: 1.3
(END)
```

After this PR:

```
Help on function lower in module pyspark.sql.functions:

lower(col)
    Converts a string expression to upper case.

    .. versionadded:: 1.5
(END)
```

Let's fix it in another PR.